src/provider/repository/stackablerepository.rs

```diff
@@ -256,8 +294,8 @@ mod tests {
         let converted_repo = StackableRepoProvider::try_from(&test_repo_crd).unwrap();
         assert_eq!(converted_repo.name, "test");
         assert_eq!(
-            converted_repo.base_url.as_str(),
-            "http://monitoring.stackable.demo:8000/"
+            converted_repo.metadata_url.as_str(),
+            "http://monitoring.stackable.demo:8000/metadata.json"
```

Comment:
Comment not related to this line, but closest I could get. 
Should we maybe also add a test to check if try_into properly returns an error for a url that cannot be a base. Currently the tests only cover that new(..) returns an error.

Reply:
In my opinion, unit tests should be white box tests. So the unit test knows that the `try_from` function creates a `StackableRepoProvider` with the already tested `new` function. It is therefore sufficient to only test that the `try_from` function is able to extract the necessary data from the given `Repository` and calls `StackableRepoProvider::new` properly. Replicating the test condition from the other unit test would also not increase the test coverage. The best solution would be to hide the option from the `try_from` function to create a `StackableRepoProvider` without calling the `new` function but that is not possible within one module.